core/src/main/resources/xsl/adapter2dot.xsl

```diff
@@ -39,7 +39,14 @@
 		</xsl:choose>
 		<xsl:value-of select="$space" />
 		<xsl:text>&quot;</xsl:text>
-		<xsl:value-of select="@firstPipe" />
+		<xsl:choose>
+			<xsl:when test="string-length(@firstPipe) > 0">
+				<xsl:value-of select="@firstPipe" />
+			</xsl:when>
+			<xsl:otherwise>
+				<xsl:value-of select="//pipe[1]/@name" />
```

Comment:
The // operator searches the whole document for (in this case) a pipe element. 
I think the following would be better:
``select="pipe[1]/@name"``